docs/core/extensions/data-redaction.md

```diff
@@ -0,0 +1,176 @@
+        // For example, here the Hmac redactor will be used for data tagged
+        // with BOTH Personal and Private (but not one without the other).
+        new DataClassificationSet(MyTaxonomy.Personal, MyTaxonomy.Private));
+});
+```
+
+Alternatively, configure it this way:
+
+```csharp
+var serviceCollection = new ServiceCollection();
+serviceCollection.AddRedaction(builder =>
+{
+    builder.SetHmacRedactor(
+        Configuration.GetSection("HmacRedactorOptions"), MyTaxonomy.Personal);
+});
+```
+
+Include this section in your JSON config file:
+
+```json
+{
+    "HmacRedactorOptions": {
+        "KeyId": 1234567890,
+        "Key": "1234567890abcdefghijklmnopqrstuvwxyz"
+    }
+}
+```
+
+- The <xref:Microsoft.Extensions.Compliance.Redaction.HmacRedactorOptions> requires its <xref:Microsoft.Extensions.Compliance.Redaction.HmacRedactorOptions.Key?displayProperty=nameWithType> and <xref:Microsoft.Extensions.Compliance.Redaction.HmacRedactorOptions.KeyId?displayProperty=nameWithType> properties to be set.
+- The `Key` should be in base 64 format and at least 44 characters long. Use a distinct key for each major deployment of a service. Keep the key material secret and rotate it regularly.
+- The `KeyId` is appended to each redacted value to identify the key used to hash the data.
+- Different key IDs mean the values are unrelated and can't be used for correlation.
+
+> [!NOTE]
+> The <xref:Microsoft.Extensions.Compliance.Redaction.HmacRedactor> is still experimental, so the preceding methods will cause the `EXTEXP0002` warningm indicating it's not yet stable.
+> To use it, add `<NoWarn>$(NoWarn);EXTEXP0002</NoWarn>` to your project file or add `#pragma warning disable EXTEXP0002` around the calls to `SetHmacRedactor`.
+
+### Configure a custom redactor
+
+To create a custom redactor, define a subclass that inherits from <xref:Microsoft.Extensions.Compliance.Redaction.Redactor>:
+
+```csharp
+public sealed class StarRedactor : Redactor
+
+public class StarRedactor : Redactor
+{
+    private const string Stars = "****";
+
+    public override int GetRedactedLength(ReadOnlySpan<char> input) => Stars.Length;
+
+    public override int Redact(ReadOnlySpan<char> source, Span<char> destination)
+    {
+        Stars.CopyTo(destination);
+
+        return Stars.Length;
+    }
+}
+```
+
+### Create a custom redactor provider
+
+The <xref:Microsoft.Extensions.Compliance.Redaction.IRedactorProvider> interface provides instances of redactors based on data classification. To create a custom redactor provider, inherit from <xref:Microsoft.Extensions.Compliance.Redaction.IRedactorProvider> as shown in the following example:
+
+```csharp
+using Microsoft.Extensions.Compliance.Classification;
+using Microsoft.Extensions.Compliance.Redaction;
+
+public sealed class StarRedactorProvider : IRedactorProvider
+{
+    private static readonly StarRedactor _starRedactor = new();
+
+    public static StarRedactorProvider Instance { get; } = new();
```

Comment:
Is the `Instance` member needed? Also, for the `GetRedactor` API when we implement a custom one like shown here, when should we consider the given `DataClassificationSet` - what's the used for?